extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -148,7 +148,17 @@ private Uni<SecurityIdentity> createSecurityIdentityWithOidcServer(RoutingContex
                 tokenUni = verifySelfSignedTokenUni(resolvedContext, request.getToken().getToken());
             }
         } else {
-            tokenUni = verifyTokenUni(resolvedContext, request.getToken().getToken());
+            if (resolvedContext.provider.verifyUserAuthSuccessWithUserInfo) {
```

Comment:
This check should only apply of the current token which is being verified is the access token, not ID token, and `AccessTokenCredential.isOpaque` returns true. IMHO this property should not apply to JWT tokens at all